File: sistemasdecompras.c
// exercicio8_sistema_compras.c
#include <stdio.h>

int main() {
    float preco, total_compra = 0.0;
    int qtd, total_itens = 0;
    int total_produtos = 0;
    
    printf("Sistema de Compras\n");
    printf("(Digite 0 no preco para finalizar)\n\n");
    
    while (1) {
        printf("Preco do produto (R$): ");
        scanf("%f", &preco);
        
        if (preco <= 0) break;
        
        printf("Quantidade: ");
        scanf("%d", &qtd);
        
        float subtotal = preco * qtd;
        total_compra += subtotal;
        total_itens += qtd;
        total_produtos++;
        
        printf("Subtotal deste produto: R$ %.2f\n\n", subtotal);
    }
    
    printf("\n--- Resumo da compra ---\n");
    printf("Total de produtos diferentes : %d\n", total_produtos);
    printf("Quantidade total de itens    : %d\n", total_itens);
    printf("Valor total da compra        : R$ %.2f\n", total_compra);
    
    if (total_produtos > 0) {
        printf("Preco medio por produto      : R$ %.2f\n", total_compra / total_produtos);
    }
    
    return 0;
}

File: SistemasDeCompras.py
total = 0
qtd_itens = 0
soma_precos = 0

print("Registro de compra (digite 0 no preço para finalizar)\n")

while True:
    try:
        preco = float(input("Preço do produto (0 para finalizar): R$ "))
        if preco == 0:
            break
        if preco < 0:
            print("Preço não pode ser negativo.")
            continue
            
        qtd = int(input("Quantidade: "))
        if qtd <= 0:
            print("Quantidade inválida.")
            continue
            
        subtotal = preco * qtd
        total += subtotal
        qtd_itens += qtd
        soma_precos += preco  # para calcular preço médio por produto diferente
        
        print(f"  → Adicionado: {qtd} × R$ {preco:.2f} = R$ {subtotal:.2f}\n")
        
    except:
        print("Valor inválido. Tente novamente.\n")

if qtd_itens > 0:
    preco_medio = total / qtd_itens
    print("\nResumo da compra:")
    print(f"Valor total da compra     : R$ {total:,.2f}")
    print(f"Quantidade total de itens : {qtd_itens}")
    print(f"Preço médio por item      : R$ {preco_medio:.2f}")
else:
    print("\nNenhum produto foi registrado.")
